### PR TITLE
[CBRD-24149] Add to test case for CBRD-24149 (11.2)

### PR DESCRIPTION
**Pull request by swi0110**
Refer to http://jira.cubrid.org/browse/CBRD-24149

Rule: http://jira.cubrid.org/browse/CBRD-24149?focusedCommentId=4760595&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-4760595

tc: http://jira.cubrid.org/browse/CBRD-24149?focusedCommentId=4760530&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-4760530

### CODE REVIEW

File: sql/_13_issues/_22_1h/cases/cbrd_24149.sql [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+);
+insert into t1 (
+    select lpad(rownum+1000,5,'0') from _db_class a, _db_class b, _db_class c limit 300
+);
+
+
+-- index prefix error
+
+-- data is not searched
+drop if exists table t2;
```
Comment (beyondykk9) [inline sql/_13_issues/_22_1h/cases/cbrd_24149.sql]: remove unnecessary key word "table"

Comment (swi0110) [reply sql/_13_issues/_22_1h/cases/cbrd_24149.sql]: thx, I will fix it.

File: sql/_13_issues/_22_1h/cases/cbrd_24149.sql [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+select /*+ recompile */ char_col from tbl1 where (char_col,char_col) in ((char_col,char_col),(char_col,char_col));
+
+-- string function (decode)
+select /*+ recompile */ char_col, decode(char_col,'a',1,'a  b',2,'ab',3,4) from tbl1;
+
+-- prepare statement
+prepare c_stmt from 'select char_col as "prepare_c" from tbl1 where char_col = ?';
+execute c_stmt using 'a';
+execute c_stmt using 'a     ';
+drop prepare char_stmt;
```
Comment (beyondykk9) [inline sql/_13_issues/_22_1h/cases/cbrd_24149.sql]: it should be changed to "drop prepare c_stmt" not to produce an error and its answer also should be modified.

File: sql/_13_issues/_22_1h/answers/cbrd_24149.answer [hunk 3]
```diff
@@ -0,0 +1,563 @@
...
+a                        
+
+===================================================
+prepare_c    
+a                        
+a                        
+a                        
+
+===================================================
+Error:-995
```
Comment (beyondykk9) [inline sql/_13_issues/_22_1h/answers/cbrd_24149.answer]: it should be modified not to porduce error.

File: sql/_13_issues/_22_1h/answers/cbrd_24149.answer [hunk 4]
```diff
@@ -0,0 +1,563 @@
...
+===================================================
+var_col    
+a       
+a  b     
+a a     
+ab     
+ab      
+
+===================================================
+Error:-493
```
Comment (beyondykk9) [inline sql/_13_issues/_22_1h/answers/cbrd_24149.answer]: it should be modified not to produce the error.

File: sql/_13_issues/_22_1h/answers/cbrd_24149.answer [hunk 5]
```diff
@@ -0,0 +1,563 @@
...
+===================================================
+0
+===================================================
+300
+===================================================
+300
+===================================================
+300
+===================================================
+Error:-493
```
Comment (beyondykk9) [inline sql/_13_issues/_22_1h/answers/cbrd_24149.answer]: it should be modified not to produce the error.

File: sql/_13_issues/_22_1h/cases/cbrd_24149.sql [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+select /*+ recompile */ char_col from tbl1 where char_col> 'a' and char_col> 'a ' and char_col<= 'a     ' using index var_idx;
+
+select /*+ recompile */ var_col from tbl1 where var_col > 'a' and var_col > 'a ' using index none;
+select /*+ recompile */ var_col from tbl1 where var_col > 'a' and var_col > 'a ' using index midx_idx1;
+select /*+ recompile */ var_col from tbl1 where var_col > 'a' and var_col > 'a ' using index var_idx;
+
+
+
+
+-- create index error
```
Comment (beyondykk9) [inline sql/_13_issues/_22_1h/cases/cbrd_24149.sql]: the comment should be modified not to produce the error.

File: sql/_13_issues/_22_1h/cases/cbrd_24149.sql [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+);
+insert into t1 (
+    select lpad(rownum+2000,5,'0') from _db_class a, _db_class b, _db_class c limit 300
+);
+insert into t1 (
+    select lpad(rownum+1000,5,'0') from _db_class a, _db_class b, _db_class c limit 300
+);
+
+
+-- index prefix error
```
Comment (beyondykk9) [inline sql/_13_issues/_22_1h/cases/cbrd_24149.sql]: the comment should be modified not to produce the error.

File: sql/_13_issues/_22_1h/cases/cbrd_24149.sql [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+    select lpad(rownum,5,'0') from _db_class a, _db_class b, _db_class c limit 300
+);
+insert into t2 (
+    select lpad(rownum+2000,5,'0') from _db_class a, _db_class b, _db_class c limit 300
+);
+
+select /*+ USE_DESC_IDX */ * from t2 where code = '00297 ' using index idx;
+select /*+ USE_DESC_IDX */ * from t2 where code = '00297 ' using index none;
+
+--prefix is not prefer
```
Comment (beyondykk9) [inline sql/_13_issues/_22_1h/cases/cbrd_24149.sql]: what's the meaning of comment?

File: sql/_13_issues/_22_1h/cases/cbrd_24149.sql [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+    select lpad(rownum+2000,5,'0') from _db_class a, _db_class b, _db_class c limit 300
+);
+insert into t1 (
+    select lpad(rownum+1000,5,'0') from _db_class a, _db_class b, _db_class c limit 300
+);
+
+
+-- index prefix error
+
+-- data is not searched
```
Comment (beyondykk9) [inline sql/_13_issues/_22_1h/cases/cbrd_24149.sql]: The comment should have meaning "data must be retrieved regardless of index use"

File: sql/_13_issues/_22_1h/cases/cbrd_24149.sql (hunk 2 above)
Comment (swi0110) [reply sql/_13_issues/_22_1h/cases/cbrd_24149.sql]: fix it

File: sql/_13_issues/_22_1h/answers/cbrd_24149.answer (hunk 3 above)
Comment (swi0110) [reply sql/_13_issues/_22_1h/answers/cbrd_24149.answer]: fix it

File: sql/_13_issues/_22_1h/cases/cbrd_24149.sql (hunk 9 above)
Comment (swi0110) [reply sql/_13_issues/_22_1h/cases/cbrd_24149.sql]: removed it

File: sql/_13_issues/_22_1h/cases/cbrd_24149.sql (hunk 8 above)
Comment (swi0110) [reply sql/_13_issues/_22_1h/cases/cbrd_24149.sql]: I change the comment

File: sql/_13_issues/_22_1h/answers/cbrd_24149.answer (hunk 4 above)
Comment (swi0110) [reply sql/_13_issues/_22_1h/answers/cbrd_24149.answer]: fix it

File: sql/_13_issues/_22_1h/cases/cbrd_24149.sql (hunk 6 above)
Comment (swi0110) [reply sql/_13_issues/_22_1h/cases/cbrd_24149.sql]: fix it

File: sql/_13_issues/_22_1h/cases/cbrd_24149.sql (hunk 7 above)
Comment (swi0110) [reply sql/_13_issues/_22_1h/cases/cbrd_24149.sql]: removed it

File: sql/_13_issues/_22_1h/answers/cbrd_24149.answer (hunk 5 above)
Comment (swi0110) [reply sql/_13_issues/_22_1h/answers/cbrd_24149.answer]: fix it

**Review by beyondykk9 — APPROVED** [review]

**Review by ssihil — APPROVED** [review]

**Review by sy052 — APPROVED** [review]

**Pull request merged**